Zapper-fi/studio
# feat(balancer-v2): add boosted tokens

## Description

to show pools like this one: https://app.beefy.com/vault/balancer-wsteth-bbaaweth
https://medium.com/balancer-protocol/balancer-launches-boosted-pools-to-increase-lp-yields-ba3daadb59f9

not totally sure if it should be separate or hacked into the pool token fetcher, as they are quite similar

## Checklist

- [X] (optional) As a contributor, my Ethereum address/ENS is: tonygao.eth

## How to test?

http://localhost:5001/apps/balancer-v2/tokens?network=arbitrum&groupIds%5B%5D=pool
after a while pools like the following should show up:
```
  {
    "key": "1838807035",
    "type": "app-token",
    "appId": "balancer-v2",
    "groupId": "pool",
    "network": "arbitrum",
    "address": "0xcba9ff45cfb9ce238afde32b0148eb82cbe63562",
    "tokens": [
      {
        "key": "2087499400",
        "type": "app-token",
        "appId": "balancer-v2",
        "groupId": "boosted",
        "network": "arbitrum",
        "address": "0xda1cd1711743e57dd57102e9e61b75f3587703da",
        "tokens": [
          {
            "type": "base-token",
            "network": "arbitrum",
            "address": "0x82af49447d8a07e3bd95bd0d56f35241523fbab1",
            "symbol": "WETH",
            "decimals": 18,
            "price": 1902.76
          }
        ],
        "symbol": "bb-a-WETH",
        "decimals": 18,
        "supply": 5192296858534828,
        "pricePerShare": [
          55650327015835650
        ],
        "price": 105889216232651440000,
        "dataProps": {
          "liquidity": 5.498082447975112e+35,
          "reserves": [
            2.889530181407593e+32
          ],
          "apy": 0,
          "isDebt": false,
          "poolId": "0xda1cd1711743e57dd57102e9e61b75f3587703da0000000000000000000003fc",
          "fee": 0.0002,
          "volume": 0
        },
        "displayProps": {
          "label": "bb-a-WETH",
          "secondaryLabel": {
            "type": "dollar",
            "value": 105889216232651440000
          },
          "images": [
            "https://storage.googleapis.com/zapper-fi-assets/tokens/arbitrum/0x82af49447d8a07e3bd95bd0d56f35241523fbab1.png"
          ],
          "statsItems": [
            {
              "label": "Liquidity",
              "value": {
                "type": "dollar",
                "value": 5.498082447975112e+35
              }
            }
          ]
        }
      },
      {
        "type": "base-token",
        "network": "arbitrum",
        "address": "0xec70dcb4a1efa46b8f2d97c310c9c4790ba5ffa8",
        "symbol": "rETH",
        "decimals": 18,
        "price": 2044.14
      }
    ],
    "symbol": "rETH-bb-a-WETH-BPT",
    "decimals": 18,
    "supply": 1297.1576661030647,
    "pricePerShare": [
      0.6179264863507693,
      0.3574593112157859
    ],
    "price": 65431751329079150000,
    "dataProps": {
      "liquidity": 8.48752978430644e+22,
      "reserves": [
        801.5480788580312,
        463.6810858634779
      ],
      "apy": 0,
      "isDebt": false,
      "poolId": "0xcba9ff45cfb9ce238afde32b0148eb82cbe635620000000000000000000003fd",
      "poolType": "ComposableStable",
      "fee": 0.0001,
      "weights": [
        0.5,
        0.5
      ],
      "volume": 0
    },
    "displayProps": {
      "label": "bb-a-WETH / rETH",
      "secondaryLabel": "100% / 0%",
      "images": [
        "https://storage.googleapis.com/zapper-fi-assets/tokens/arbitrum/0x82af49447d8a07e3bd95bd0d56f35241523fbab1.png",
        "https://storage.googleapis.com/zapper-fi-assets/tokens/arbitrum/0xec70dcb4a1efa46b8f2d97c310c9c4790ba5ffa8.png"
      ],
      "statsItems": [
        {
          "label": "Liquidity",
          "value": {
            "type": "dollar",
            "value": 8.48752978430644e+22
          }
        }
      ]
    }
  },
```


note: get an error from balancer, not related to this pr:
```
[Nest] 23605  - 06/02/2023, 11:22:03 PM   ERROR [CacheOnIntervalService] @CacheOnInterval error init for ArbitrumBalancerV2FarmContractPositionFetcher#getPositions: Multicall call failed for 0x6c34d77a57226f9df6ec476b20913350832ebfec:reward_data(address): call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="reward_data(address)", data="0x000000000000000000000000c38c5f97b34e175ffd35407fc91a937300e338600000000000000000000000000000000000000000000000000000000064828dbe000000000000000000000000000000000000000000000000000001dd5d4a4b7800000000000000000000000000000000000000000000000000000000647a9f2a00000000000000000000000000000000000000000000000000008d90659e3e8a", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.7.0) (decode)
```
